validator/client/validator.go

```diff
@@ -133,10 +133,10 @@ func (v *validator) RoleAt(slot uint64) pb.ValidatorRole {
 	if v.assignment == nil {
 		return pb.ValidatorRole_UNKNOWN
 	}
-	if v.assignment.AttesterSlot == slot {
-		return pb.ValidatorRole_ATTESTER
-	} else if v.assignment.ProposerSlot == slot {
+	if v.assignment.ProposerSlot == slot {
```

Comment:
Please add a comment here since this is now an important distinction. Something like: `A validator could be assigned to propose and attest in the same slot. Proposing takes priority.`

Also, if a validator is selected to do both, should they do both?

Reply:
#1613